dash-pipeline/bmv2/stages/metering_update.p4

```diff
@@ -82,27 +80,19 @@ control metering_update_stage(
     }
     
     apply {
-        if (meta.meter_policy_en == 1) {
+        meta.meter_class = meta.meter_context.meter_class_or & meta.meter_context.meter_class_and;
+
+        // If the meter class is 0 from the SDN policies, we go through the metering policy.
+        if (meta.meter_class == 0) {
             meter_policy.apply();
             meter_rule.apply();
         }
-
-        {
-            if (meta.meter_policy_en == 1) {
-                meta.meter_class = meta.policy_meter_class;
-            } else {
-                meta.meter_class = meta.route_meter_class;
-            }
-            if ((meta.meter_class == 0) || (meta.mapping_meter_class_override == 1)) {
-                meta.meter_class = meta.mapping_meter_class;
-            }
-        }
-
         meter_bucket.apply();
+
         if (meta.direction == dash_direction_t.OUTBOUND) {
-            UPDATE_COUNTER(meter_bucket_outbound, meta.meter_bucket_index);
+            UPDATE_COUNTER(meter_bucket_outbound, meta.meter_class);
         } else if (meta.direction == dash_direction_t.INBOUND) {
-            UPDATE_COUNTER(meter_bucket_inbound, meta.meter_bucket_index);
+            UPDATE_COUNTER(meter_bucket_inbound, meta.meter_class);
         }
```

Comment:
We should also check if meter class is not 0 for this section of code.

Reply:
You are right! This is a good one!

Reply:
It is updated now. : D